packages/react/src/ReactContext.js

```diff
@@ -67,7 +67,46 @@ export function createContext<T>(
     $$typeof: REACT_PROVIDER_TYPE,
     _context: context,
   };
-  context.Consumer = context;
+
+  if (__DEV__) {
+    // A separate object, but proxies back to the original context object for
+    // backwards compatibility. It has a different $$typeof, so we can properly
+    // warn for the incorrect usage of Context as a Consumer.
+    context.Consumer = {
+      $$typeof: REACT_CONTEXT_TYPE,
+      _context: context,
+      get _calculateChangedBits() {
```

Comment:
I think getters don't work on www, can you rewrite this as `Object.defineProperty` calls?
Worth checking tho

Reply:
Really? They work in IE9 the last time I checked.

Reply:
I changed over to `Object.defineProperties`. It's a shame Flow has such bad support for them though :(